test/v5/L2Deposit.ts

```diff
@@ -0,0 +1,357 @@
+import { expect } from "chai"
+import { BigNumber, ContractReceipt } from "ethers"
+import { ethers } from "hardhat"
+import { Network, isNetwork } from "@network"
+import {
+    L2Deposit,
+    L2ArbitrumDepositData,
+    L2OptimismDepositData,
+    SignatureType,
+    encodingHelper,
+    signingHelper,
+} from "@src/v5"
+import { dealTokenAndApprove } from "@test/utils/balance"
+import { EXPIRY } from "@test/utils/constant"
+import { contextSuite } from "@test/utils/context"
+import {
+    parseLogsByName,
+    deployERC1271Wallet,
+    deployERC1271WalletETHSign,
+} from "@test/utils/contract"
+
+enum L2Identifier {
+    Arbitrum,
+    Optimism,
+}
+
+const zeroAddress = "0x" + "0".repeat(40)
+
+if (isNetwork(Network.Goerli)) {
+    contextSuite("L2Deposit", ({ wallet, network, token, tokenlon }) => {
+        // Configure the default deposit information
+        const defaultDeposit: L2Deposit = {
+            l2Identifier: 0,
+            l1TokenAddr: zeroAddress,
+            l2TokenAddr: zeroAddress,
+            sender: wallet.user.address,
+            recipient: wallet.user.address,
+            amount: 100,
+            salt: 1234,
+            expiry: EXPIRY,
+            data: "0x", // It should be a hexadecimal value
+        }
+        // Declare the gas data for the Arbitrum bridge
+        const arbitrumGasData: L2ArbitrumDepositData = {
+            maxSubmissionCost: ethers.utils.parseUnits("1", "mwei"),
+            maxGas: ethers.utils.parseUnits("1", "mwei"),
+            gasPriceBid: ethers.utils.parseUnits("1", "gwei"),
+        }
+        // Declare the gas data for the Optimism bridge
+        const optimismGasData: L2OptimismDepositData = {
+            l2Gas: ethers.utils.parseUnits("1", "mwei"),
+        }
+
+        /* Test for the Arbitrum bridge  */
+
+        describe("Deposit on the Arbitrum bridge", () => {
+            // Test for the EOA
+            it(`Should sign and encode a valid deposit for an EOA to use on the Arbitrum bridge.`, async () => {
+                const deposit = {
+                    // Override the default deposit data for the Arbitrum bridge
+                    ...defaultDeposit,
+                    l2Identifier: L2Identifier.Arbitrum,
+                    l1TokenAddr: token.USDT.address,
+                    l2TokenAddr: token.USDTForArbitrumBridge.address,
+                    data: encodingHelper.encodeL2ArbitrumDepositData(
+                        wallet.user.address,
+                        arbitrumGasData,
+                    ),
+                }
+                // Obtain a specific amount of tokens and approve their to be used by the AllowanceTarget contract
+                await dealTokenAndApprove(
+                    wallet.user,
+                    tokenlon.AllowanceTarget,
+                    deposit.l1TokenAddr,
+                    defaultDeposit.amount,
```

Comment:
Use the deposit constructed in test instead of the default (I found other tests have the same issue, can be updated all together.)

Reply:
No problem!
But I didn't find the same issue in any other test files except the [LimitOrder.ts](https://github.com/consenlabs/tokenlon-contracts-lib-js/blob/LABS-1180/L2Deposit/test/v5/LimitOrder.ts#L60-L90) file.
The approved amount parameters in this LimitOrder file are being used before the order constructed are declared, right?